gordo_components/builder/build_model.py

```diff
@@ -150,10 +169,62 @@ def build_model(
     }
 
     metadata["model"].update(_get_metadata(model))
-
     return model, metadata
 
 
+def get_metrics_dict(metrics_list, y):
+    """
+        Given a list of metrics that accept a true_y and pred_y as inputs this returns
+        a dictionary with keys in the form '{score}-{tag_name}' for each given target tag
+        and '{score}-tag-mean' corresponding to the average score across all target tags and folds,
+        and values being the callable make_scorer(metric_wrapper(score)).
+        Note: score in {score}-{tag_name} is a sklearn's score function name with '_' replaced by '-' and tag_name
+        corresponds to given tag name with ' ' replaced with '-'.
+
+        Parameters
+        ----------
+        metrics: list
+            List of sklearn score functions
+        y: pd.DataFrame
+            Target data
+
+
+        Returns
+        -------
+            dict
+        """
+
+    def _score_factory(metric=r2_score, index=0):
+        def _score_per_tag(y_true, y_pred):
+            # This function extracts the score for each given target_tag to
+            # use as scoring argument in sklearn cross_validate, as the scoring
+            # must return a single value.
+            return metric(y_true.values[:, index], y_pred[:, index])
```

Comment:
Seems like we're making some fairly inflexible assumptions here.
1. That `y_true` _will_ be a dataframe
2. That `y_pred` _will_ be a numpy array

Perhaps we can accept either for both and do some `hasattr(..., 'values')` or something similar to be a bit more flexible?